src/pages/index.js

```diff
@@ -5,52 +5,348 @@ import PopupWithForm from "../components/PopupWithForm.js";
 );
 const cardListEl = document.querySelector(".cards__list");
-// export const profileForm = document.querySelector(
-//   "#profile-edit-modal .modal__form"
-// );
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  jobSelector: ".profile__description",
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "acb9f423-9227-4882-a1e0-a6492f6bb248",
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cardData]) => {
+    userInfo.setUserInfo(userData);
+    section = new Section(
+      {
+        items: cardData,
+        renderer: (cardData) => {
+          renderCard(cardData);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+// api
+//   .getInitialCards()
+//   .then((cardData) => {
+//     if (cardData) {
+//       section = new Section(
+//         {
+//           items: cardData,
+//           renderer: renderCard,
+//         },
+//         cardListEl
+//       );
+//       section.renderItems();
+//     }
+//   })
+//   .catch((err) => console.log("Error loading cards:", err));
+
+let user;
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+// avatar edit modal
+
+const avatarChangeModal = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarChangeSubmit
+);
+avatarChangeModal.setEventListeners();
+
+// ---------------------------------------
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+
+function handleAvatarChangeSubmit(inputValues) {
+  // event.preventDefault();
+  avatarSubmitButton.textContent = "Saving...";
+
+  // const avatarForm = event.currentTarget;
+  const avatarUrl = inputValues.avatar;
+  console.log("avatarUrl:", avatarUrl);
+  // const avatarUrl = event.target.avatar.value;
+
+  if (avatarUrl) {
+    api
+      .updateAvatar(avatarUrl)
+      .then((updatedUser) => {
+        userInfo.setUserInfo(updatedUser);
+        avatarChangeModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        avatarSubmitButton.textContent = "Save";
+      });
+  } else {
+    console.error("Avatar URL is not defined");
+    avatarSubmitButton.textContent = "Save"; // Ensure the button text is reset
+  }
+}
+
+const avatarFormElement = document.querySelector("#avatar-form");
+if (avatarFormElement) {
+  avatarFormElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const inputValues = new FormData(event.target);
+    handleAvatarChangeSubmit(Object.fromEntries(inputValues));
+  });
+}
+
+// Event Listener for opening avatar modal
+avatarImage.addEventListener("click", () => {
+  submitButtonSelector.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.